### ISSUE TITLE
google_maps_flutter: Location Service Icon on the status bar is turned ON after disposing GoogleMap

### ISSUE THREAD

**Issue by yousifAlneamy** [flutter/flutter#56154]
When opening `GoogleMap` in a screen, the GPS/location icon is shown on the status bar. However, when leaving the screen (press back button), the icon is still there which means location services is still turned on and seems like the app is still using the device location services.

<img src="https://user-images.githubusercontent.com/21126886/80846490-11db2d80-8c15-11ea-92a0-a33bfa525e46.png" alt="screenshot" width="340">

```
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';
import 'package:location/location.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Google Maps Demo',
      home: InitScreen(),
    );
  }
}

class InitScreen extends StatefulWidget {
  @override
  _InitScreenState createState() => _InitScreenState();
}

class _InitScreenState extends State<InitScreen> {
  @override
  void initState() {
    super.initState();
    // Im using this only one time then commenting it just to enable location services on the first usage
    // getLocationPermission();
  }

  getLocationPermission() async {
    var location = new Location();
    try {
      location.requestPermission(); //to lunch location permission popup
    } on PlatformException catch (e) {
      if (e.code == 'PERMISSION_DENIED') {
        print('Permission denied');
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        child: Center(
          child: RaisedButton(
            child: Text('Open Map'),
            onPressed: () => Navigator.of(context).push(MaterialPageRoute(
              builder: (context) => MapSample(),
            )),
          ),
        ),
      ),
    );
  }
}

class MapSample extends StatefulWidget {
  @override
  State<MapSample> createState() => MapSampleState();
}

class MapSampleState extends State<MapSample> {
  Completer<GoogleMapController> _controller = Completer();

  static final CameraPosition _kGooglePlex = CameraPosition(
    target: LatLng(37.42796133580664, -122.085749655962),
    zoom: 14.4746,
  );

  static final CameraPosition _kLake = CameraPosition(
      bearing: 192.8334901395799,
      target: LatLng(37.43296265331129, -122.08832357078792),
      tilt: 59.440717697143555,
      zoom: 19.151926040649414);

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      body: GoogleMap(
        myLocationButtonEnabled: true,
        myLocationEnabled: true,
        mapToolbarEnabled: true,
        zoomControlsEnabled: false,
        mapType: MapType.normal,
        initialCameraPosition: _kGooglePlex,
        onMapCreated: (GoogleMapController controller) {
          _controller.complete(controller);
        },
      ),
      // floatingActionButton: FloatingActionButton.extended(
      //   onPressed: _goToTheLake,
      //   label: Text('To the lake!'),
      //   icon: Icon(Icons.directions_boat),
      // ),
    );
  }

  Future<void> _goToTheLake() async {
    final GoogleMapController controller = await _controller.future;
    controller.animateCamera(CameraUpdate.newCameraPosition(_kLake));
  }
}

```

**flutter doctor -v**

```
[✓] Flutter (Channel beta, v1.17.0-3.3.pre, on Mac OS X 10.15.4 19E287, locale en-IQ)
    • Flutter version 1.17.0-3.3.pre at /Users/yousif/development/flutter
    • Framework revision 0da1ab0922 (3 days ago), 2020-04-28 11:02:34 -0700
    • Engine revision 376ad6a64b
    • Dart version 2.8.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/yousif/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.8.4

[!] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 1 category.
```

**Comment by TahaTesser** [comment]
This issue also reproducible using official example without location package.
Location icon is visible after closing the app and using regular Google Maps from Play Store, removes this icon when app is in background

<img width="335" alt="Screenshot 2020-05-04 at 5 37 55 PM" src="https://user-images.githubusercontent.com/48603081/80964237-1dee0780-8e2e-11ea-8207-a3398f420161.png">


<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, v1.18.0-8.0.pre, on Mac OS X 10.15.4 19E287, locale
    en-GB)
    • Flutter version 1.18.0-8.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision e0c63cd35e (10 days ago), 2020-04-23 22:29:01 -0400
    • Engine revision d2ec21221e
    • Dart version 2.9.0 (build 2.9.0-3.0.dev b0d35855d8)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (4 available)
    • AOSP on IA Emulator • emulator-5554 • android-x86    • Android 9 (API 28)
      (emulator)
    • macOS               • macOS         • darwin-x64     • Mac OS X 10.15.4
      19E287
    • Web Server          • web-server    • web-javascript • Flutter Tools
    • Chrome              • chrome        • web-javascript • Google Chrome
      81.0.4044.129

• No issues found!
``` 

</details>

**Comment by Danaru87** [comment]
On Android, passing App to background don't hide icon.

![gps icon stay](https://user-images.githubusercontent.com/5873203/81064839-b0e47b80-8eda-11ea-9751-bb4cfd9a28c0.png)

**Comment by yousifAlneamy** [comment]
Fixed by `0.5.28+1`
